src/env_globals.ts

```diff
@@ -113,12 +113,23 @@ export const detectEnvironmentByGlobals = (config: OxlintConfigOrOverride) => {
 
     let matches = search.filter(
       (entry) =>
-        // @ts-ignore -- we already checked for undefined
+        // @ts-expect-error -- we already checked for undefined
         entry in config.globals &&
-        // @ts-ignore -- filtering makes the key to any
+        // @ts-expect-error -- filtering makes the key to any
         normalizeGlobValue(config.globals[entry]) === entries[entry]
     );
-    if (search.length === matches.length) {
+
+    // For browser and node, we allow a match if >97% of keys match
+    // This lets us handle version differences in globals package where
+    // there's a difference of just a few extra/removed keys.
+    // Do not do any other envs, otherwise things like es2024 and es2026
+    // would match each other.
+    const useThreshold = env === 'browser' || env === 'node';
+
+    if (
+      (useThreshold && matches.length / search.length >= 0.97) ||
```

Comment:
I am not sure if we want to have a threshold in percentage here. I thought more about "max 10 items missing".
Should we add the missing global as disabled, like `"IDBRecord": "off"`?
Would love to hear a second opinion :)

Reply:
And as for setting the missing globals to false, I'm not sure if that's necessarily worth the trouble? My main argument is just that most users are probably upgrading the globals package without much thought anyway. I doubt it's a concern if migrating to oxlint gets them a few more globals. They would've gotten those same extra values next time they upgraded the globals package anyway.